### PR TITLE
You can stick duct tape to more than just paper now

### LINKED ISSUE #10337: Couple of tape-related runtimes
Just putting these here for reference; I plan to add some functionality to sticky tape, when I'll fix these two.

Caused if you spawn in a piece of sticky tape and then examine it

```
[2018-11-24T18:36:17] Runtime in tape.dm,91: Cannot execute null.examine().
  proc name: examine (/obj/item/ducttape/examine)
  usr: Goodwin Joyce (citinited) (/mob/living/carbon/human)
  usr.loc: The floor (125,125,1) (/turf/simulated/floor/wood)
```

Caused when attacking a nonhuman with tape roll
```
[2018-11-24T18:33:22] Runtime in tape.dm,21: undefined proc or verb /mob/living/simple_animal/bot/secbot/beepsky/check has mouth().

  proc name: attack (/obj/item/stack/tape_roll/attack)
  usr: Goodwin Joyce (citinited) (/mob/living/carbon/human)
  usr.loc: The floor (80,142,1) (/turf/simulated/floor/plasteel)
  src: the tape roll (/obj/item/stack/tape_roll)
  src.loc: Goodwin Joyce (/mob/living/carbon/human)
  call stack:
  the tape roll (/obj/item/stack/tape_roll): attack(Officer Beepsky (/mob/living/simple_animal/bot/secbot/beepsky), Goodwin Joyce (/mob/living/carbon/human))
  Officer Beepsky (/mob/living/simple_animal/bot/secbot/beepsky): attackby(the tape roll (/obj/item/stack/tape_roll), Goodwin Joyce (/mob/living/carbon/human), "icon-x=16;icon-y=8;left=1;scre...")
  Officer Beepsky (/mob/living/simple_animal/bot/secbot/beepsky): attackby(the tape roll (/obj/item/stack/tape_roll), Goodwin Joyce (/mob/living/carbon/human), "icon-x=16;icon-y=8;left=1;scre...")
  Officer Beepsky (/mob/living/simple_animal/bot/secbot/beepsky): attackby(the tape roll (/obj/item/stack/tape_roll), Goodwin Joyce (/mob/living/carbon/human), "icon-x=16;icon-y=8;left=1;scre...")
  Officer Beepsky (/mob/living/simple_animal/bot/secbot/beepsky): attackby(the tape roll (/obj/item/stack/tape_roll), Goodwin Joyce (/mob/living/carbon/human), "icon-x=16;icon-y=8;left=1;scre...")
  the tape roll (/obj/item/stack/tape_roll): melee attack chain(Goodwin Joyce (/mob/living/carbon/human), Officer Beepsky (/mob/living/simple_animal/bot/secbot/beepsky), "icon-x=16;icon-y=8;left=1;scre...")
  Goodwin Joyce (/mob/living/carbon/human): ClickOn(Officer Beepsky (/mob/living/simple_animal/bot/secbot/beepsky), "icon-x=16;icon-y=8;left=1;scre...")
  Officer Beepsky (/mob/living/simple_animal/bot/secbot/beepsky): Click(the floor (79,141,1) (/turf/simulated/floor/plasteel), "mapwindow.map", "icon-x=16;icon-y=8;left=1;scre...")
```

### LINKED ISSUE #10374: Duct Tape taping jackboots instead of silencing it
**Problem Description**:
Tried to silence jackboots with the duct tape and it just added a tape to the boot as if I were to stick it somewhere.

Didn't try to actually tape the boot to a wall, but well, wasn't supposed to do that anyway.
And the boots didn't get silenced either.

screenshot_2

### PR DESCRIPTION
**Pull request by Citinited**
image

**What does this PR do:**
This PR adds the ability to stick duct tape to any item, not just paper. For example, you can now tape pipes to the floor to keep them in place when there's a hull breach! To remove the tape, there's an action button.

Removes `/obj/item/ducttape` - did you know when you stick duct tape to paper, it transforms the stuck piece of paper into an `obj/item/ducttape`? This makes for a whole load of unintended behaviour, for example being unable to set fire to ducttaped paper.

Fixes #10337.
Fixes #10374.

~~TODO: #9743~~ wait fuck that, that bug is obnoxious and infuriating

Yes I'm well aware that you can make things look weird by applying ducttape to odd angles of items. It's a feature:tm:

**Images of sprite/map changes (IF APPLICABLE):**
image

:cl:
tweak: You can now tape much more than just paper!
add: You can now order a crate full of sticky tape
tweak: Maximum amount in a tape roll is now 25 (up from 10).
/:cl:

### CODE REVIEW

**Comment by Mitchs98** [comment]
Can you duct tape airlocks closed?
tenor

**Comment by Citinited** [comment]
No, I mean duct tape is pretty strong but not *that* strong 😄

**Comment by Ty-Omaha** [comment]
Duct tape should be removed we are in the future

Everyone should be using flex tape

And instead of metal foam it needs to be flex seal

**Comment by Mitchs98** [comment]
> No, I mean duct tape is pretty strong but not _that_ strong 
> 
> Maybe science could research glue that can hold two bulkheads together or something though

Simple solution is to rename it flex seal and add airlock sealing properties.

**Comment by KasparoVy** [comment]
I know we're going to see some bloody hilarious tape shenanigans and I'm excited

**Comment by Citinited** [comment]
Apply duct tape to "KICK ME" paper, stick directly onto the captain's butt

**Comment by KasparoVy** [comment]
Absolutely cover a dude in taped papes

Hahaha oh my God

**Comment by davidchan13** [comment]
We're not going to create some tape singularity by sticking tape to tape now are we?

**Comment by MarsM0nd** [comment]
> create some tape singularity

What if you can just do that, tape the singularity in place. Who needs containment anyway?
Don't think that's possible, though, looks to me that it will only work on obj/items, and the singularity is just an obj/singularity, but best to check for such things.

**Comment by taukausanake** [comment]
> What if you can just do that

No fun allowed.
But it should be

**Comment by KasparoVy** [comment]
Playtest 👽👌

**Comment by Citinited** [comment]
I have done, it's easy enough when you can just check the paths of everything 😄

**Comment by KasparoVy** [comment]
Nah I mean testmerge style where we can let the devious and creative of us use/break it in unexpected ays

**Comment by Citinited** [comment]
Haha well enjoy the testmerge happening later!

**Comment by KasparoVy** [comment]
Taping footwear currently applies the sticky as well as silences footsteps, and using wirecutters to remove the tape only toecuts footwear and doesn't actually remove the tape. Haven't been able to test shoes yet

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -415,6 +419,12 @@ var/global/image/fire_overlay = image("icon" = 'icons/goonstation/effects/fire.d
 //The default action is attack_self().
 //Checks before we get to here are: mob is alive, mob is not restrained, paralyzed, asleep, resting, laying, item is on the mob.
 /obj/item/proc/ui_action_click(mob/user, actiontype)
+	if(actiontype == /datum/action/item_action/remove_tape)
```
Comment (variableundefined) [inline code/game/objects/items.dm]: Erm. 

Is there a better way to implement this on the item_action instead of this? This seems horribly snowflakey.

Comment (Citinited) [reply code/game/objects/items.dm]: Oh right, haha. There almost certainly will be!

**Comment by craftxbox** [comment]
taping jackboots silences them AND puts a peice of tape on them. bug?

**Comment by KasparoVy** [comment]
Taping footwear that can be silenced period does this, and can be resolved by simply clicking the action button to remove the excess tape

**Comment by Citinited** [comment]
I've got a fix for the shoe thing, I'll commit once I've done a bit more work on this

**Comment by variableundefined** [comment]
Once the shoe is fixed and the snowflakey code is addressed I can merge it in (Assuming no further problems are discovered)

File: code/datums/supplypacks.dm [hunk 2]
```diff
@@ -1341,6 +1341,14 @@ var/list/all_supply_groups = list(supply_emergency,supply_security,supply_engine
 	cost = 15
 	containername = "bureaucracy crate"
 
+/datum/supply_packs/misc/tape
```
Comment (KasparoVy) [inline code/datums/supplypacks.dm]: _You think you hear a faint quacking from within..._

Comment (Citinited) [reply code/datums/supplypacks.dm]: 🤔

Comment (taukausanake) [reply code/datums/supplypacks.dm]: Make the tape crate have a random chance to contain a rubber ducky

Comment (Citinited) [reply code/datums/supplypacks.dm]: This... why didn't I think of this before

Comment (Citinited) [reply code/datums/supplypacks.dm]: Change committed 📯 

Also there's a bug right now where if you open / close a crate with a squeaky thing inside, the squeaky thing will squeak.

So if you open this you might get a nice quack to greet you 😄

**Comment by variableundefined** [comment]
Some items don't show a duct tape overlay properly:
Potted plants
Folder
Beaker 
etc.

**Comment by Mitchs98** [comment]
Jackboots can no longer be silenced still. Normal shoes can be however :thinking:

**Comment by Citinited** [comment]
I'll do a final pass over the items listed above, then should be good to go

**Comment by Citinited** [comment]
Okay I think I hunted down most of the most commonly-used items. I can't be fucked finding any more, I'm sure people can create issues when they discover some more 😄

**Comment by variableundefined** [comment]
Are jackboots working now?

**Comment by Citinited** [comment]
Yah I changed how the footsteps work slightly - they can now be silenced with tape properly.

**Comment by KasparoVy** [comment]
Brilliant

**Pull request merged**